inc/VcfReader.hpp

```diff
@@ -0,0 +1,242 @@
+     * - If `HIGH_QUAL_ONLY` is true and the call has low quality, no field after QUAL is loaded.
+     * - If `PASS_ONLY` is true and the call is not PASS, no field after FILTER is loaded.
+     * - If the call is not of a supported type, no field after INFO is loaded.
+     * - If the call is shorter than `MIN_SV_LENGTH`, no field after INFO is loaded.
+     * - If there are more samples than `N_SAMPLES_IN_VCF`, no sample after the maximum is loaded.
+     *
+     * @param tmp_* reused temporary space.
+     */
+    void set(ifstream& stream, string& tmp_buffer_1, string& tmp_buffer_2, pair<uint8_t, uint8_t>& tmp_pair);
+
+    /**
+     * @return TRUE iff the record passes all the constraints set at construction time.
+     */
+    bool passes_constraints() const;
+
+    /**
+     * Appends to `stream` a string representation of the object. No terminator is added at the end.
+     */
+    void print(ostream& stream) const;
+
+    /**
+     * Remark: the procedure handles the case where there are two copies of the chromosome but the GT field contains
+     * just one value (this can happen with CNVs or when the sample is a haploid reference).
+     *
+     * @param sample sample ID (the first sample has ID zero);
+     * @param out output pair that stores the only GT haplotype ID in `first`, or the two haplotype IDs in `first` and
+     * `second`, depending on the ploidy of `chrom`; -1 = missing haplotype;
+     * @param tmp_buffer reused temporary space;
+     * @return number of GT haplotypes in `sample` (zero iff `sample` is invalid).
+     */
+    uint8_t get_gt(uint32_t sample, pair<int8_t,int8_t>& out, string& tmp_buffer) const;
+
+private:
+    /**
+     * Constraints specified by the user at construction time.
+     *
+     * Remark: PASS_ONLY means FILTER=PASS or FILTER='.'
+     */
+    bool PASS_ONLY, HIGH_QUAL_ONLY, SKIP_SAMPLES;
+    uint32_t MIN_SV_LENGTH, N_SAMPLES_IN_VCF;
+    float MIN_QUAL;
+    double MIN_N_HAPLOTYPES_ALT, MIN_N_HAPLOTYPES_NONMISSING;
+
+    /**
+     * Sets `sv_type` using `ref`, `alt` and `info`, which are assumed to be already set.
+     *
+     * @param tmp_buffer reused temporary space.
+     */
+    void set_sv_type(string& tmp_buffer);
+
+    /**
+     * Sets `sv_length` using the following VCF fields, in order:
+     * - INFO.SVLEN if it exists;
+     * - INFO.END if it exists;
+     * - non-symbolic ALT if it exists.
+     *
+     * Remark: `ref`, `alt` and `info` are assumed to be already set.
+     *
+     * @param tmp_buffer reused temporary space.
+     */
+    void set_sv_length(string& tmp_buffer);
+
+    /**
+     * Main logic of `set()`.
+     *
+     * @param tmp_* reused temporary space;
+     * @return TRUE iff some VCF fields were skipped.
+     */
+    bool set_field(const string& field, uint32_t field_id, bool high_qual_only, float min_qual, bool pass_only, uint32_t min_sv_length, bool skip_samples, ifstream& stream, string& tmp_buffer, pair<uint8_t, uint8_t>& tmp_pair);
```

Comment:
Same issue here as with `tmp_buffer` in `get_gt()`

Reply:
I kept `tmp_buffer` exposed here, since this function is private and its calls are orchestrated by `set()` (public).